plugin/src/functions/submitTask/handler.ts

```diff
@@ -2,24 +2,56 @@ import { functionTaskTable } from '@libs/dynamodb';
 import { APIGatewayEvent } from 'aws-lambda';
 import sqs from '../../libs/awsSqs';
 import { v4 } from 'uuid';
+import { TaskMessageBody } from '@libs/types';
+
+/**
+ * Takes the body attribute that is passed in by the API Gateway proxy event
+ * and makes sure that it is either parsed to an object, or if it is falsy
+ * (e.g. null/undefined/empty string) it will return undefined.
+ *
+ * The AWS types say that it should always be a string, but it seems like
+ * it can also come through as null. If it doesn't match the type contract
+ * we should probably be extra cautious and check for all kinds of falsy value.
+ *
+ * @param body the body attribute passed in by an API Gateway proxy event
+ * @returns either the parsed object or undefined
+ */
+const tryExtractBody = (body: string): unknown | undefined => {
+  const coercedBody = body ?? '';
+  if (coercedBody.trim().length > 0) {
+    return JSON.parse(body);
+  }
+  return undefined;
+};
 
 const submitTask = async (event: APIGatewayEvent) => {
   try {
     const functionName = event.pathParameters['functionName'];
 
     const taskId = v4();
 
-    const apiUrl = event.headers['X-Forwarded-Proto'] + '://' + event.headers['Host'] + '/' + event.requestContext['stage'];
+    const apiUrl =
```

Comment:
future: I wonder if this should be passed into the handler by the plugin rather than re-created here.